xdsl/dialects/llvm.py

```diff
@@ -95,6 +97,66 @@ def is_typed(self):
         return not isinstance(self.type, NoneAttr)
 
 
+@irdl_op_definition
+class GEPOp(Operation):
+    name = "llvm.getelementptr"
+
+    ptr: Annotated[Operand, LLVMPointerType]
+
+    ssa_indices: Annotated[VarOperand, IntegerType]
+
+    elem_type: OptOpAttr[Attribute]
+
+    rawConstantIndices: OpAttr[DenseArrayBase]
+
+    inbounds: OptOpAttr[UnitAttr]
+
+    result: Annotated[OpResult, LLVMPointerType]
+
+    @staticmethod
+    def get(ptr: SSAValue | Operation,
+            result_type: Attribute
```

Comment:
```suggestion
            result_type:
```

Why both Attribute and LLVMPointerType?

Reply:
When the callee passes a none LLVMPointerType, it immediately throws an error. This way we can catch it and return a ValueError. I could make it just Attribute and just add a bit more logic for opaque type instead of setting it as the default here?

Reply:
I don't understand, the only input it can handle without raising errors is `LLVMPointerType`, so it seems that we should communicate that through the type system, unless I'm missing something

Reply:
Ok, fixed!